### ISSUE TITLE
Volume Video Issue (iOS Programming/ Objective-c)

### ISSUE THREAD

**Issue by pourang** [wit-ai/wit#137]
When I am using "Wit" in my current "iOS Programming" Objective-c project that has a video introduction, the volume of video goes down  as low as possibe. The volume value is 1.0 but it plays like volume = 0.3. Is there any side effect of using Wit with playing music or video?

**Comment by mristuccia** [comment]
Hi pourang,

only to say that I have the same problem!
In my app sometimes I call the Wit stop() method and play some audio, then restart Wit listening. When I play the audio on my device (iPhone 4s) the volume is very low even if it is set at its maximum (1.0).

**Comment by burkaygur** [comment]
Same issue here! Developers, please take a look? 

What I have observed is that the sound switches to the headset headphones (where you hear the phone calls), that is why it looks like the volume is going down, but in fact it is switching to the wrong headphone output! Using this in your viewDidLoad fixes the issue of switching the speakers, but the method is deprecated in iOS 8.

```
[[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayAndRecord error:nil];
UInt32 audioRouteOverride = kAudioSessionOverrideAudioRoute_Speaker;
AudioSessionSetProperty(kAudioSessionProperty_OverrideAudioRoute, sizeof(audioRouteOverride), &audioRouteOverride);
```

So now the second issue is that the audio gets stopped completely when the recording is stopped. I think that is because the system wide AVSession looks like it is being stopped when the recording is stopped. Looking at the source code, that seems like to be the case. I am not sure about the solution, but the SDK has to make sure that it is only stopping its own AVSession, and not everything. The error seems to be:

```
AVAudioSession.mm:623: -[AVAudioSession setActive:withOptions:error:]: Deactivating an audio session that has running I/O. All I/O should be stopped or paused prior to deactivating the audio session.
```

Please help on this one!

**Comment by mristuccia** [comment]
Hi burkaygur,

thank you for your temporary fix, I hope that the developers here will hear us and patch the code.

About the second issue, I'm able to stop the audio recording, play a sound, restart recording and so on without loosing the audio output. With or without your fix.

**Comment by Catacola** [comment]
Hi all,

We recently released [version 4.0.0 of the SDK](https://github.com/wit-ai/wit-ios-sdk/releases/tag/4.0.0). In order to cover the wide range of possible uses for Wit, we've handed control of the AudioSession over to the developer.

Documentation for this version [is up](https://wit.ai/docs/ios/4.0.0/quickstart). I'd suggest paying particular attention to section 5 to see the changes you'll need to make.
